Mathlib/Analysis/SpecialFunctions/PolarCoord.lean

```diff
@@ -211,3 +215,87 @@ protected theorem lintegral_comp_polarCoord_symm (f : ℂ → ℝ≥0∞) :
   simp_rw [measurableEquivRealProd_symm_polarCoord_symm_apply]
 
 end Complex
+
+section Pi
+
+open ENNReal MeasureTheory MeasureTheory.Measure
+
+variable {ι : Type*}
+
+open ContinuousLinearMap in
+/-- The derivative of `polarCoord.symm` on `ι → ℝ × ℝ`, see `hasFDerivAt_pi_polarCoord_symm`. -/
+noncomputable def fderivPiPolarCoordSymm (p : ι → ℝ × ℝ) : (ι → ℝ × ℝ) →L[ℝ] ι → ℝ × ℝ :=
+  pi fun i ↦ (fderivPolarCoordSymm (p i)).comp (proj i)
+
+theorem injOn_pi_polarCoord_symm :
+    Set.InjOn (fun p (i : ι) ↦ polarCoord.symm (p i)) (Set.univ.pi fun _ ↦ polarCoord.target) :=
+  fun _ hx _ hy h ↦ funext fun i ↦ polarCoord.symm.injOn (hx i trivial) (hy i trivial)
+    ((funext_iff.mp h) i)
+
+theorem abs_fst_of_mem_pi_polarCoord_target {p : ι → ℝ × ℝ}
+    (hp : p ∈ (Set.univ.pi fun _ : ι ↦ polarCoord.target)) (i : ι) :
+    |(p i).1| = (p i).1 :=
+  abs_of_pos ((Set.mem_univ_pi.mp hp) i).1
+
+variable [Fintype ι]
+
+theorem hasFDerivAt_pi_polarCoord_symm (p : ι → ℝ × ℝ) :
+    HasFDerivAt (fun x i ↦ polarCoord.symm (x i)) (fderivPiPolarCoordSymm p) p := by
+  rw [fderivPiPolarCoordSymm, hasFDerivAt_pi]
+  exact fun i ↦ HasFDerivAt.comp _ (hasFDerivAt_polarCoord_symm _) (hasFDerivAt_apply i _)
+
+theorem det_fderiv_pi_polarCoord_symm (p : ι → ℝ × ℝ) :
```

Comment:
```suggestion
theorem det_fderivPiPolarCoordSymm (p : ι → ℝ × ℝ) :
```
That's the point of the naming convention: when you refer to an object (which has no underscore in its name, by design), then in the lemma name you can put his name unbroken, and use underscores to separate the different objects. In practice, it works very well (much better than the mixed convention we had in Lean 3).